From 8f9b9f7a9d3e4b1c2a8f9b9f7a9d3e4b1c2a8f9b Mon Sep 17 00:00:00 2001
From: ServerJDK Developer <dev@serverjdk.org>
Date: Wed, 24 Jun 2026 00:00:00 +0000
Subject: [PATCH 1/2] Optimize G1GC defaults for game servers

This patch modifies the default G1GC heuristics to favor throughput and 
aggressive young generation sizing, which is ideal for workloads like 
Minecraft that allocate millions of short-lived objects per second.
---
 src/hotspot/share/gc/g1/g1_globals.hpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/hotspot/share/gc/g1/g1_globals.hpp b/src/hotspot/share/gc/g1/g1_globals.hpp
index abcdef12345..fedcba54321 100644
--- a/src/hotspot/share/gc/g1/g1_globals.hpp
+++ b/src/hotspot/share/gc/g1/g1_globals.hpp
@@ -62,7 +62,7 @@
           "The default value of this flag must be < 1.0")                   \
           range(0.0, 1.0)                                                   \
                                                                             \
-  product(uintx, G1NewSizePercent, 5,                                       \
+  product(uintx, G1NewSizePercent, 35,                                      \
           "Percentage (0-100) of the heap size to use as default "          \
           "minimum young gen size.")                                        \
           range(0, 100)                                                     \
@@ -72,7 +72,7 @@
           "maximum young gen size.")                                        \
           range(0, 100)                                                     \
                                                                             \
-  product(uintx, MaxGCPauseMillis, 200,                                     \
+  product(uintx, MaxGCPauseMillis, 50,                                      \
           "Adaptive size policy maximum GC pause time goal in millisecond, "\
           "or (G1 Only) the maximum GC time per MMU time slice")            \
           range(1, max_uintx - 1)                                           \
@@ -83,7 +83,7 @@
           "within the pause time goal")                                     \
           range(0, 100)                                                     \
                                                                             \
-  product(size_t, G1HeapRegionSize, 0,                                      \
+  product(size_t, G1HeapRegionSize, 8388608,                                \
           "Size of the G1 regions.")                                        \
           range(0, 32*M)                                                    \
           constraint(G1HeapRegionSizeConstraintFunc,AfterMemoryInit)        \
-- 
2.39.2
